### DISCUSSION TITLE
signData for Java/Kotlin

### DISCUSSION THREAD

**Discussion by malenalbc** [trustwallet/wallet-core#1252] [Q&A] [answered]
I'm using 2.5.3 for signing a [forge result](https://tezos.gitlab.io/007/rpc.html#post-block-id-helpers-forge-operations) with Tezos as part of the common flow for transacting. The forge result comes as an hexadecimal string of variable length. Is there some way to sign it with my `HDWallet` private key directly?

Reviewing the `AnySigner.java` class my options are three methods:

`sign(Message input, CoinType coin, Parser<T> parser)`
`nativeSign(byte[] data, int coin)` 
`signJSON(String json, byte[] key, int coin)`

The first option would be my way to go, but `Tezos.SigningInput` only accepts a `privateKey` and an `operationList`. Second one doesn't seem to use the private key anywhere and the third fails with an `invalid_argument` error.

I see the `Signer` class has a `signData` method that does exactly what I'd need. Is there any way to call it either directly  or indirectly from Java/Kotlin? Thanks!

**Comment by hewigovens** [comment, marked as answer]
I think class `PrivateKey` can help you, just construct the data and pass it to `sign` method; to get the private key from `HDWallet`, call `getKeyForCoin`
